Development/sdp/sdp_grammar.cpp

```diff
@@ -714,6 +714,54 @@ namespace sdp
                 {
                     sdp::attributes::mediaclk,
                     string_converter // sorry, cannot summon the energy
+                },
+                {
+                    sdp::attributes::extmap,
+                    {
+                        [](const web::json::value& v) {
+                            std::string s;
+                            s += number_converter.format(v.at(sdp::fields::local_id));
+                            if (v.has_field(sdp::fields::direction)) s += "/" + string_converter.format(v.at(sdp::fields::direction));
+                            s += " " + string_converter.format(v.at(sdp::fields::uri));
+                            if (v.has_field(sdp::fields::extensionattributes)) s += " " + string_converter.format(v.at(sdp::fields::extensionattributes));
+                            return s;
+                        },
+                        [](const std::string& s) {
+                            auto v = web::json::value::object(keep_order);
+                            size_t pos = (!s.empty() && ' ' == s.front()) ? 1 : 0;
+                            v[sdp::fields::local_id] = number_converter.parse(substr_find(s, pos, bst::regex{ R"([ \/])" }));
+                            if (s.at(pos - 1) == '/') v[sdp::fields::direction] = string_converter.parse(substr_find(s, pos, " "));
+                            v[sdp::fields::uri] = string_converter.parse(substr_find(s, pos, " "));
+                            if (std::string::npos != pos) v[sdp::fields::extensionattributes] = string_converter.parse(substr_find(s, pos));
+                            return v;
+                        }
+                    }
+                },
+                {
+                    sdp::attributes::hkep,
+                    {
+                        [](const web::json::value& v) {
+                            std::string s;
+                            s += number_converter.format(v.at(sdp::fields::port));
```

Comment:
It looks more like SP is not allowed here as well, that would allow us to just reuse `object_converter`.

https://www.rfc-editor.org/rfc/rfc5234#appendix-B.1 says that `HEXDIG` is uppercase. Should we transform the IDs from JSON representation to uppercase while generating SDP representation?

Reply:
You're right, `HEXDIG` is uppercase, but TR-10-5 doesn't include ABNF that could state clearly `HEXDIG`, so it's not completely clear.

I assume **NodeId** an NMOS Node `id`? In which case, should that follow NMOS and be lowercase?

How about **PortId**, what spec defines this 5 octet thing?

Maybe these are questions for VSF...